array.c

```diff
@@ -3168,9 +3199,24 @@ rb_ary_to_h(VALUE ary)
 
 /*
  *  call-seq:
- *     ary.to_ary -> ary
+ *    array.to_ary -> self or new_array
  *
- *  Returns +self+.
+ *  When <tt>self.instance_of?(Array)</tt>, returns +self+:
+ *    a = [:foo, 'bar', 2]
+ *    a.instance_of?(Array) # => true
+ *    a1 = a.to_ary
+ *    a1.equal?(a) # => true # Returned self
+ *
+ *  Otherwise, when <tt>self.kind_of?(Array)</tt>,
+ *  returns a new \Array containing the actual elements (not copies) of +self+:
+ *    class MyArray < Array; end
+ *    a = MyArray.new(['foo', 'bar', 'two'])
+ *    a.instance_of?(Array) # => false
+ *    a.kind_of?(Array) # => true
+ *    a1 = a.to_ary
+ *    a1.class # => Array # Not MyArray
```

Comment:
Is this correct? As long I can see `to_ary` returns an object with the same class that self:

```
ruby -e "class MyArray < Array; end; puts MyArray.new([]).to_ary.class"
MyArray
```

`to_ary` does sightly different work than `to_a` which effectively returns an `Array` instance:

```ruby
ruby -e "class MyArray < Array; end; puts MyArray.new([]).to_a.class"
Array

```

Reply:
Oops.  Fixed in next push.  Thanks.